riscv/insns/lpad.h

```diff
@@ -0,0 +1,6 @@
+if (ZICFILP_xLPE(STATE.v, STATE.prv) && STATE.elp == elp_t::LP_EXPECTED) {
+  software_check(pc % 4 == 0 &&
+                 ((READ_REG(7) >> 12) == static_cast<uint64_t>(insn.u_imm()) || insn.u_imm() == 0LL),
+                 LANDING_PAD_FAULT);
```

Comment:
replace with as only bits 31:12 should be compared. Also note that `u_imm()` already returns the value left shifted by 12.
```
 software_check(pc % 4 == 0 &&
                 ((READ_REG(7) & 0xFFFFF000) == (insn.u_imm() & 0xFFFFF000) || insn.u_imm() == 0LL),
                 LANDING_PAD_FAULT);

```

Reply:
Fixed